pkg/util/errutil/errors.go

```diff
@@ -1,9 +1,184 @@
 package errutil
 
 import (
+	"errors"
 	"fmt"
 )
 
+// Base represents the static information about a specific error.
+// Always use NewBase to create new instances of Base.
+type Base struct {
+	// Because Base is typically instantiated as a package or global
+	// variable, having private members reduces the probability of a
+	// bug messing with the error base.
+	reason        StatusReason
+	messageID     string
+	publicMessage string
+	logLevel      LogLevel
+}
+
+// NewBase initializes a Base that is used to construct Error:s.
+// The reason is used to determine the status code that should be
+// returned for the error, and the msgID is passed to the caller
+// to serve as the base for user facing error messages.
+//
+// msgID should be structured as component.error-brief, for example
+//   login.failed-authentication
+//   dashboards.validation-error
+//   dashboards.uid-already-exists
```

Comment:
🤔 Nice enough to make it so that `NewBase` cannot be used directly for package level variable declarations or write a new linter?

I'd rather keep that out of scope for the initial implementation.